examples/tensorflow/inference/crnn/README.md

```diff
@@ -0,0 +1,119 @@
+
+For more details please refer to https://docs.ucloud.cn/ai/uai-inference/index
+
+### 2.1 Writing Service Code
+We provide the example service code in inference.py. We defined ocrModel which derived from TFAiUcloudModel. 
+
+In ocrModel we only need to implement load_model funcs and execute funcs:
+
+1. load_model(self),which loads the given object detection model.
+    ``` 
+    def load_model(self):
+		sess = tf.Session()
+		x = tf.placeholder(dtype=tf.float32, shape=[1, 32, 100, 3], name='input')
+		'''
+		1 define model
+		'''
+		net = crnn_model.ShadowNet(phase='Test', hidden_nums=256, layers_nums=2, seq_length=25, num_classes=37)
+		with tf.variable_scope('shadow'):
+			net_out = net.build_shadownet(inputdata=x)
+		decodes, _ = tf.nn.ctc_beam_search_decoder(inputs=net_out, sequence_length=25*np.ones(1), merge_repeated=False)
+		
+		sess_config = tf.ConfigProto()
+		sess = tf.Session(config=sess_config)
+		'''
+		2 load model from self.model_dir
+		'''
+		saver = tf.train.Saver()
+		params_file = tf.train.latest_checkpoint(self.model_dir)
+		saver.restore(sess=sess, save_path=params_file)
+		'''
+		3 Register ops into self.output dict.
+		  So func execute() can get these ops
+		'''
+		self.output['sess'] = sess
+		self.output['x'] = x
+		self.output['y_'] = decodes
+    ```
+2. execute(self, data, batch_size) which will do the inference. The django server will invoke ocrModel->execute when it receives requests. It will buffer requests into a array named 'data' when possible. In execute(), it first preprocesses the input 'data' array by invoking self.resize_image() for each data[i] and merge all preprocessed images into one batch, then recognize the text. It formats result into list object. (You can format it into json also)
+    ```
+    	def execute(self, data, batch_size):	
+    		sess = self.output['sess']
+    		x = self.output['x']
+    		y_ = self.output['y_']
+    		decoder = data_utils.TextFeatureIO()
+    		ret = []
+    		for i in range(batch_size):
+    			'''
+    			1 load data 
+    			'''
+    			image = np.array(Image.open(data[i]))
+    			image = cv2.resize(image, (100, 32))
+    			image = np.expand_dims(image, axis=0).astype(np.float32)
+    			'''
+    			2 inference
+    			'''
+    			preds = sess.run(y_, feed_dict={x: image})
+    			preds = decoder.writer.sparse_tensor_to_str(preds[0])[0]+'\n'
+    			ret.append(preds)
+    		return ret
+    ```
+
+### 2.2 Define the Config File
+We need to provide the config file to tell the UAI Inference system to get the basic information to load  model. The config file should include following info:
+
+1  "exec" tells which file is used as the entry-point of the user-defined inference logic and which main class is used. <br>
+2  "tensorflow" tells which model related info should be parsed by UAI Inference system.
+
+
+
+## 3 Preparing model
+You can use the model you trained on UAI Train Platform.We provide a crnn model in inference/checkpoint_dir/.
```

Comment:
You can use the model you trained on UAI Train Platform. You can also down it according to checkpoint_dir/intro.txt